external-api/api-tests/v1/middleware/api-rate-limiting.api-test.ts

```diff
@@ -1,73 +1,70 @@
+import * as dotenv from 'dotenv';
+import axios from 'axios';
 import { generateApp } from '../../../src/generateApp';
 import { api } from '../../api';
+import { ENTITY_TYPE, UKEF_ID, USER } from '../../../src/constants';
 
-const mockResponse = {
-  status: 202,
-  data: {
-    id: '7b7475ca5c984a808cb56abdc9b75a61',
-    statusQueryGetUri:
-      'http://localhost:7072/runtime/webhooks/durabletask/instances/7b7475ca5c984a808cb56abdc9b75a61?taskHub=numbergenerator&connection=Storage&code=Xm80X2xSRGxNKd9YtLYDBkynz49/knUL5lI/7KauDj0/duawyQb06A==',
-    sendEventPostUri:
-      'http://localhost:7072/runtime/webhooks/durabletask/instances/7b7475ca5c984a808cb56abdc9b75a61/raiseEvent/{eventName}?taskHub=numbergenerator&connection=Storage&code=Xm80X2xSRGxNKd9YtLYDBkynz49/knUL5lI/7KauDj0/duawyQb06A==',
-    terminatePostUri:
-      'http://localhost:7072/runtime/webhooks/durabletask/instances/7b7475ca5c984a808cb56abdc9b75a61/terminate?reason={text}&taskHub=numbergenerator&connection=Storage&code=Xm80X2xSRGxNKd9YtLYDBkynz49/knUL5lI/7KauDj0/duawyQb06A==',
-    rewindPostUri:
-      'http://localhost:7072/runtime/webhooks/durabletask/instances/7b7475ca5c984a808cb56abdc9b75a61/rewind?reason={text}&taskHub=numbergenerator&connection=Storage&code=Xm80X2xSRGxNKd9YtLYDBkynz49/knUL5lI/7KauDj0/duawyQb06A==',
-    purgeHistoryDeleteUri:
-      'http://localhost:7072/runtime/webhooks/durabletask/instances/7b7475ca5c984a808cb56abdc9b75a61?taskHub=numbergenerator&connection=Storage&code=Xm80X2xSRGxNKd9YtLYDBkynz49/knUL5lI/7KauDj0/duawyQb06A==',
-  },
-};
+dotenv.config();
 
-jest.mock('../../../src/v1/controllers/durable-functions-log.controller');
+const { RATE_LIMIT_THRESHOLD } = process.env;
 
-jest.mock('axios', () =>
-  jest.fn(() => {
-    return Promise.resolve(mockResponse);
-  }),
-);
+const rateLimit = 2;
 
-describe('api rate limiting', () => {
-  const rateLimit = 2;
+const body = {
+  entityType: ENTITY_TYPE.DEAL,
+  dealId: '1234',
+};
+const mockSuccessfulResponse = {
+  status: 200,
+  data: [
+    {
+      id: 12345678,
+      maskedId: UKEF_ID.TEST,
+      type: 1,
+      createdBy: USER.DTFS,
+      createdDatetime: '2024-01-01T00:00:00.000Z',
+      requestingSystem: USER.DTFS,
+    },
+  ],
+};
 
-  const requestBody = {
-    dealId: 111,
-    dealType: 'dealType',
-    entityId: 222,
-    entityType: 'deal',
-    user: { id: 'userId' },
-  };
+axios.post = jest.fn().mockResolvedValue(mockSuccessfulResponse);
 
+describe('api rate limiting', () => {
   let originalRateLimitThreshold: string | undefined;
-  let app;
-  let post: any;
   let sendRequestTimes: any;
 
   beforeEach(() => {
-    originalRateLimitThreshold = process.env.RATE_LIMIT_THRESHOLD;
+    // Initialise environment variable
+    originalRateLimitThreshold = RATE_LIMIT_THRESHOLD;
     process.env.RATE_LIMIT_THRESHOLD = rateLimit.toString();
```

Comment:
seeing as it's only used here I'd probably just put '2' here instead of separating it out into a variable that you have to scroll up to read. Not too bothered about this one though happy for you to just resolve if you disagree

Reply:
turns out I can't read, ignore me!